### PR TITLE
Adds top level 'infra' configuration section to the azure.yaml schema

### PR DESCRIPTION
**Pull request by wbreza**
Updates the `azure.yaml` JSON schema for the new infra provider configuration options.

### CODE REVIEW

File: schemas/v1.0/azure.yaml.json [hunk 1]
```diff
@@ -27,6 +27,31 @@
...
             }
         },
+        "infra": {
+            "type": "object",
+            "additionalProperties": true,
+            "properties": {
+                "provider":{
+                    "type": "string",
+                    "title": "Type of infrastructure provisioning provider",
+                    "description": "Optional. When specified will set the infrastructure provisioning provider used to provision the Azure resources for the application. (Default: bicep)",
```
Comment (weikanglim) [inline schemas/v1.0/azure.yaml.json:37]: "When specified will set" seems redundant. Maybe move it to the end, if we want to keep it? 

```suggestion
                    "description": "Optional. The infrastructure provisioning provider used to provision the Azure resources for the application. (Default: if unset bicep)",
```


>

File: schemas/v1.0/azure.yaml.json [hunk 2]
```diff
@@ -27,6 +27,31 @@
...
+                    "enum": [
+                        "bicep",
+                        "terraform"
+                    ]
+                },
+                "path": {
+                    "type": "string",
+                    "title": "Path to the location that contains Azure provisioning templates",
```
Comment (weikanglim) [inline schemas/v1.0/azure.yaml.json:45]: Does this support both relative and full path?

File: schemas/v1.0/azure.yaml.json [hunk 3]
```diff
@@ -27,6 +27,31 @@
...
+                    "description": "Optional. When specified will override the path to the Azure provisioning templates for the specified provider. (Default: infra)"
+                },
+                "module": {
```
Comment (weikanglim) [inline schemas/v1.0/azure.yaml.json:48]: Thinking out loud: is allowing a custom `module` entry-point a useful feature? I wonder if using `main` always is better for consistency.

**Review by karolz-ms — APPROVED** [review]

File: schemas/v1.0/azure.yaml.json (hunk 2 above)
Comment (weikanglim) [reply schemas/v1.0/azure.yaml.json:45]: Maybe the same question can be imposed: do we think this is a useful enough feature to break convention? Being opinionated about repo structure might not be a bad thing.

File: schemas/v1.0/azure.yaml.json [hunk 4]
```diff
@@ -27,6 +27,31 @@
...
+                "provider":{
+                    "type": "string",
+                    "title": "Type of infrastructure provisioning provider",
+                    "description": "Optional. When specified will set the infrastructure provisioning provider used to provision the Azure resources for the application. (Default: bicep)",
+                    "enum": [
+                        "bicep",
+                        "terraform"
+                    ]
+                },
+                "path": {
```
Comment (weikanglim) [inline schemas/v1.0/azure.yaml.json:43]: Maybe the same question can be imposed: do we think this is a useful enough feature to break convention? Being opinionated about repo structure might not be a bad thing.

**Review by weikanglim — COMMENTED** [review]
Everything looks simple enough and good for checking in. I am curious to see if @jongio has any thoughts here though.

File: schemas/v1.0/azure.yaml.json (hunk 2 above)
Comment (wbreza) [reply schemas/v1.0/azure.yaml.json:45]: It is relative paths.  Will update the wording to call this out.

File: schemas/v1.0/azure.yaml.json (hunk 3 above)
Comment (wbreza) [reply schemas/v1.0/azure.yaml.json:48]: We already support this for the `service` components.  We'll have our convention by default but allows devs to use their own pattern.

File: schemas/v1.0/azure.yaml.json (hunk 4 above)
Comment (wbreza) [reply schemas/v1.0/azure.yaml.json:43]: I believe there was an feature request issue filed for this.  Will search for it.

**Review by jongio — APPROVED** [review]

File: schemas/v1.0/azure.yaml.json [hunk 5]
```diff
@@ -27,6 +27,33 @@
...
+                    "description": "Optional. The infrastructure provisioning provider used to provision the Azure resources for the application. (Default: bicep)",
+                    "enum": [
+                        "bicep",
+                        "terraform"
+                    ]
+                },
+                "path": {
+                    "type": "string",
+                    "title": "Path to the location that contains Azure provisioning templates",
+                    "description": "Optional.The relative folder path to the Azure provisioning templates for the specified provider. (Default: infra)"
```
Comment (weikanglim) [inline schemas/v1.0/azure.yaml.json:48]:
```suggestion
                    "description": "Optional. The relative folder path to the Azure provisioning templates for the specified provider. (Default: infra)"
```

**Review by weikanglim — APPROVED** [review]

File: schemas/v1.0/azure.yaml.json [hunk 6]
```diff
@@ -27,6 +27,33 @@
...
+                },
+                "module": {
```
Comment (ellismg) [inline schemas/v1.0/azure.yaml.json:50]: Just wondering - Does this have a semantics for terraform?  IIRC they just load all the `.tf` files from the folder, unlike bicep where there is a single entrypoint.

**Review by ellismg — APPROVED** [review]

File: schemas/v1.0/azure.yaml.json (hunk 6 above)
Comment (wbreza) [reply schemas/v1.0/azure.yaml.json:50]: Correct.

**Pull request merged**